lib/ansible/cli/adhoc.py

```diff
@@ -71,7 +73,18 @@ def post_process_args(self, options):
     def _play_ds(self, pattern, async_val, poll):
         check_raw = context.CLIARGS['module_name'] in C.MODULE_REQUIRE_ARGS
 
-        mytask = {'action': {'module': context.CLIARGS['module_name'], 'args': parse_kv(context.CLIARGS['module_args'], check_raw=check_raw)},
+        module_args_raw = context.CLIARGS['module_args']
+        module_args = None
+        if module_args_raw and module_args_raw.startswith('{') and module_args_raw.endswith('}'):
+            try:
+                module_args = from_yaml(module_args_raw.strip(), json_only=True)
```

Comment:
not sure we need json_only .. the yaml shorthand is more flexible and should allow people to use with a LOT less quote mangling

Reply:
The reason why I went strict json is I don't want something like `-a '{{ variable }}'` to be read as YAML. Using strict json makes it a bit more stringent on what it will accept for the json data and have it fallback to the existing kv parser.

Reply:
That already fails the YAML parser as a dictionary is not a hashable key

Reply:
While it does seem to fail with pyyaml it is being parsed by yamllint so maybe in the future pyyaml might load it. I just kept it as json to be on the safe side.

Reply:
Which in this case is valid yaml, it's just pyyaml that doesn't like it. I have no idea if that's going to change in the future. I prefer to be safe rather than sorry and someone else can implement yaml support in the future if they wish to pursue that path.

Reply:
for it to be valid a dictionary  would need to be a valid key in a python dictionary